backend/api/publishing.py

```
@@ -107,61 +311,234 @@ async def save_scenario_draft(
+                debug_log(f"DUPLICATE PREVENTION: Found existing scenario ID {existing_scenario.id}, updating instead of creating new one")
+                # Update the existing scenario instead of creating a new one
+                existing_scenario.description = actual_ai_result.get("description", "")
+                existing_scenario.challenge = actual_ai_result.get("description", "")
+                existing_scenario.learning_objectives = actual_ai_result.get("learning_outcomes", [])
+                existing_scenario.student_role = actual_ai_result.get("student_role", "Business Analyst")
+                existing_scenario.completion_status = actual_ai_result.get("completion_status", {})
+                existing_scenario.grading_config = actual_ai_result.get("grading_config", {})
+                existing_scenario.rubric_title = actual_ai_result.get("rubric_title")
+                existing_scenario.rubric_criteria = actual_ai_result.get("rubric_criteria")
+                existing_scenario.rubric_performance_levels = actual_ai_result.get("rubric_performance_levels")
+                existing_scenario.updated_at = datetime.utcnow()
```

Ruff (0.14.1)

`datetime.datetime.utcnow()` used

(DTZ003)